src/pages/groupComparison/pathwayMapper/GroupComparisonPathwayMapper.tsx

```diff
@@ -0,0 +1,158 @@
+import * as React from 'react';
+import PathwayMapperTable, {
+    IPathwayMapperTable,
+} from '../../../shared/lib/pathwayMapper/PathwayMapperTable';
+import { observer } from 'mobx-react';
+import autobind from 'autobind-decorator';
+import { computed, makeObservable, observable } from 'mobx';
+import { Row } from 'react-bootstrap';
+
+import PathwayMapper from 'pathway-mapper';
+
+import 'pathway-mapper/dist/base.css';
+import 'cytoscape-panzoom/cytoscape.js-panzoom.css';
+import 'cytoscape-navigator/cytoscape.js-navigator.css';
+import { AlterationEnrichmentRow } from 'shared/model/AlterationEnrichmentRow';
+import { ComparisonGroup } from '../GroupComparisonUtils';
+import { GenesSelection } from 'pages/resultsView/enrichments/GeneBarPlot';
+import { DefaultTooltip } from 'cbioportal-frontend-commons';
+import GroupComparisonStore from '../GroupComparisonStore';
+import { GeneOptionLabel } from 'pages/resultsView/enrichments/EnrichmentsUtil';
+
+interface IGroupComparisonPathwayMapperProps {
+    genesOfInterest: AlterationEnrichmentRow[];
+    activeGroups: ComparisonGroup[] | undefined;
+    genomicData: AlterationEnrichmentRow[];
+    store: GroupComparisonStore;
+}
+
+@observer
+export default class GroupComparisonPathwayMapper extends React.Component<
+    IGroupComparisonPathwayMapperProps
+> {
+    clearMessage: (() => void) | undefined;
+
+    constructor(props: IGroupComparisonPathwayMapperProps) {
+        super(props);
+        makeObservable(this);
+
+        // @ts-ignore
+        import(/* webpackChunkName: "pathway-mapper" */ 'pathway-mapper').then(
+            (module: any) => {
+                this.PathwayMapperComponent = (module as any)
+                    .default as PathwayMapper;
+            }
+        );
+    }
+
+    @observable.ref PathwayMapperComponent:
+        | PathwayMapper
+        | undefined = undefined;
+    @observable isGeneSelectionPopupVisible: boolean = false;
+    @observable _genesOfInterest: string[] | undefined = undefined;
+
+    @computed get selectedGenes() {
+        return (
+            this._genesOfInterest ||
+            this.props.store.genesWithMaxFrequency.map(g => g.hugoGeneSymbol)
+        );
+    }
+
+    @computed get genesOfInterest(): AlterationEnrichmentRow[] {
+        return this.props.store.alterationEnrichmentRowData.filter(gene => {
```

Comment:
We have to get more disciplined about what we pass in as a prop and what we take off the store.  The question is where we are awaiting the promise that underlies this data.   If this whole component is only to be used on the group comparison page, then maybe it makes sense to JUST pass in the store and await anything specific to this component here.  @onursumer you'll need to advise on this.

Reply:
We need to await these 2 data: this.alterationsEnrichmentData.isComplete && this.alterationsEnrichmentAnalysisGroups.isComplete to compute genesOfInterest. We are awaiting the promise in the GroupComparisonPage.tsx. Await operation is done here because other tabs also need to use store, not only Pathways tab.